src/common/_modules/private_endpoint/main.tf

```diff
@@ -0,0 +1,22 @@
+resource "azurerm_private_endpoint" "this" {
+  for_each = local.private_endpoints
+
+  name                = "${var.project}-${each.key}-pep-01"
```

Comment:
I'd put the enumeration part in the locals file, as a resource could have multiple private endpoints. Especially resources in other subscriptions as we try to create a private endpoint per consumer